Emit output what to do next. This fixes #921

hannesm opened the pull request:
It gives some hints to the user what they should do next.

The only thing slightly worrying is that the cram test cases have the log message two times - any hints why this is the case?

----

We could after the `build` improve depending on the target, if we would have access to the target. But I have a tough time to figure out how to pass this information through.

So, this is ready for review :D

palainp reviewed (approved):
Dear @hannesm , thank you for that PR it should be helpful for newcomers :)
I tried (but failed) to find why we call configure two times.

The first one seems to comes from `let* () = Engine.configure i jobs in` in lib.ml that calls `let* () = Device.configure dev info in` in engine.ml which calls configure that prints the message.

I however have not found yet the second caller. Maybe it can be possible to add an optional do_not_print_msg defaulted to false, and call with true the first time?

hannesm commented:
rebased, fixed, and opened #1594 for the future. NB: getting a callstack is pretty easy:

```OCaml
  let bt = Printexc.get_callstack 20 in
  Logs.info (fun m -> m "backtrace: %s" (Printexc.raw_backtrace_to_string bt));
```
